src/repos/FlyerRepo.ts

```diff
@@ -178,17 +179,19 @@ const refreshTrendingFlyers = async (): Promise<Flyer[]> => {
   });
 
   // Get new trending Flyers
-  const flyers = await FlyerModel.aggregate()
-    // Get a sample of random Flyers
-    .sample(100)
-    // Get Flyers after 30 days ago
-    .match({
-      date: {
-        $gte: new Date(
-          new Date().setDate(new Date().getDate() - MAX_NUM_DAYS_OF_TRENDING_ARTICLES),
-        ),
-      },
-    });
+  const flyers = await (
+    await FlyerModel.aggregate()
+      // sort flyers by trendiness
+      .sort({ trendiness: 'desc' })
+      // Only get flyers for events that start in the next few days
+      .match({
+        startDate: {
+          $lte: new Date(
+            new Date().setDate(new Date().getDate() + MAX_NUM_DAYS_OF_TRENDING_ARTICLES),
+          ),
+        },
+      })
+  ).slice(0, MAX_NUM_OF_TRENDING_FLYERS);
```

Comment:
I'm not sure exactly how this resolves but I don't think you need the double `await` here, you should only need the one outside the parenthesis in order to do what you want